### ISSUE TITLE
Expression grouping with parentheses not respected

### ISSUE THREAD

**Issue by valeriangalliat** [marko-js/marko#543]
### Context

I'm trying to compile the following template:

```html
<script marko-init>
  const states = {
    foo: 'bar'
  };
</script>
<p>${(state_names[state] || state)[0].toUpperCase()}</p>
```

But the grouping around `(state_names[state] || state)` is not respected in the output.

### Expected Behavior

```js
// ...

  return function render(data, out) {
    out.w("<p>" +
      escapeXml((state_names[state] || state)[0].toUpperCase()) +
      "</p>");
  };
}

// ...
```

### Actual Behavior

```js
// ...

  return function render(data, out) {
    out.w("<p>" +
      escapeXml(state_names[state] || state[0].toUpperCase()) +
      "</p>");
  };
}

// ...
```

See how the grouping around `state_names[state] || state` is dropped.

### Your Environment

* Version used:
* Marko 3.14.0
* Node.js 6.9.2
* macOS Sierra (10.12.2)

**Comment by patrick-steele-idem** [comment]
Thanks for reporting the problem. I'm investigating now.

**Comment by patrick-steele-idem** [comment]
Resolved. New version published with fix: `marko@3.14.1`

This bug had already been fixed in Marko v4.

Thanks again for reporting the problem.

**Comment by valeriangalliat** [comment]
Thanks a lot!
